cpp/src/gcylon/sorting/deserialize.cpp

```diff
@@ -0,0 +1,221 @@
+        disp.push_back(displacements_per_buffer.at(i).at(tableNo));
+    }
+    return disp;
+}
+
+bool allZero(std::vector<int32_t> &buffer_sizes) {
+    for (int i: buffer_sizes) {
+        if (i != 0) {
+            return false;
+        }
+    }
+    return true;
+}
+
+cylon::Status TableDeserializer::deserialize(std::vector<std::shared_ptr<cylon::Buffer>> &received_buffers,
+                                                 std::vector<std::vector<int32_t>> &displacements_per_buffer,
+                                                 std::vector<std::vector<int32_t>> &buffer_sizes_per_table,
+                                                 std::vector<std::unique_ptr<cudf::table>> &received_tables) {
+
+    int number_of_tables = buffer_sizes_per_table.size();
+
+    for (int i = 0; i < number_of_tables; ++i) {
+        if (allZero(buffer_sizes_per_table.at(i))) {
+            continue;
+        }
+        std::vector<int32_t> disp = displacementsPerTable(displacements_per_buffer, i);
+        std::unique_ptr<cudf::table> outTable = deserializeTable(received_buffers,
+                                                                 disp,
+                                                                 buffer_sizes_per_table.at(i));
+        received_tables.push_back(std::move(outTable));
+    }
+
+    return cylon::Status::OK();
+}
+
+std::unique_ptr<cudf::column> constructColumn(std::shared_ptr<rmm::device_buffer> data_buffer,
+                                              std::shared_ptr<rmm::device_buffer> mask_buffer,
+                                              std::shared_ptr<rmm::device_buffer> offsets_buffer,
+                                              cudf::data_type dt,
+                                              int32_t num_rows) {
+
+    std::unique_ptr<cudf::column> clmn = nullptr;
+
+    // if it is non-string column
+    if (dt.id() != cudf::type_id::STRING) {
+        if (mask_buffer->size() == 0) {
+            clmn = std::make_unique<cudf::column>(dt, num_rows, std::move(*data_buffer));
+        } else {
+            clmn = std::make_unique<cudf::column>(dt,
+                                                  num_rows,
+                                                  std::move(*data_buffer),
+                                                  std::move(*mask_buffer));
+        }
+        // if it is a string column
+    } else {
+        // construct chars child column
+        auto cdt = cudf::data_type{cudf::type_id::INT8};
+        auto chars_column = std::make_unique<cudf::column>(cdt, data_buffer->size(), std::move(*data_buffer));
+
+        auto odt = cudf::data_type{cudf::type_id::INT32};
+        auto offsets_column = std::make_unique<cudf::column>(odt, num_rows + 1, std::move(*offsets_buffer));
+
+        std::vector<std::unique_ptr<cudf::column>> children;
+        children.emplace_back(std::move(offsets_column));
+        children.emplace_back(std::move(chars_column));
```

Comment:
```suggestion
        std::vector<std::unique_ptr<cudf::column>> children{ std::move(offsets_column), std::move(chars_column)};
```

Reply:
this does not compile.

Reply:
yes! my bad! 
https://stackoverflow.com/a/25827463/4116268